### ISSUE TITLE
Process continues remaining after job execution

### ISSUE THREAD

**Issue by papix** [key-amb/perl5-App-Koyomi#13]
In our environment(Mac OS X, Perl 5.20.2, App::Koyomi v0.5.2), App::Koyomi run correctly job.
However, job process will continue to exist.

I will show the config file, job list, log of koyomi and process list:

```
[log]
debug   = true
console = true
file    = true

[worker]
interval_minutes = 1
minimum_interval_seconds = 30

[schedule]
update_interval_seconds = 120

[job]
lock_ttl_seconds = 25

[datasource.module]
job       = "Teng"
semaphore = "None"

[datasource.connector]
dsn      = "dbi:mysql:database=koyomi_trial;host=127.0.0.1;port=3306"
user     = "root"
password = ""

[debug.worker]
sleep_seconds = 5
```

```
$ koyomi-cli list -c config/default.toml
2015-07-15T18:15:40 [DEBUG] bless( {'datasource' => {'connector' => {'dsn' => 'dbi:mysql:database=koyomi_trial;host=127.0.0.1;port=3306','password' => '','user' => 'root'},'module' => {'job' => 'Teng','semaphore' => 'None'}},'debug' => {'worker' => {'sleep_seconds' => 5}},'job' => {'lock_ttl_seconds' => 25},'log' => {'console' => 'true','debug' => 'true','file' => 'true'},'schedule' => {'update_interval_seconds' => 120},'worker' => {'interval_minutes' => 1,'minimum_interval_seconds' => 30}}, 'App::Koyomi::Config' ) at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Config.pm line 30
.--------------------------------------------------------------.
| id | user | command            | Y | m | d | H | M | weekday |
+----+------+--------------------+---+---+---+---+---+---------+
|  3 |      | echo $(date) alive | * | * | * | * | * | *       |
'----+------+--------------------+---+---+---+---+---+---------'
```

```
$ koyomi -c config/default.toml
2015-07-15T18:11:32 [DEBUG] bless( {'datasource' => {'connector' => {'dsn' => 'dbi:mysql:database=koyomi;host=127.0.0.1;port=3306','password' => '','user' => 'root'},'module' => {'job' => 'Teng','semaphore' => 'None'}},'debug' => {'worker' => {'sleep_seconds' => 5}},'job' => {'lock_ttl_seconds' => 25},'log' => {'console' => 'true','debug' => 'true','file' => 'true'},'schedule' => {'update_interval_seconds' => 120},'worker' => {'interval_minutes' => 1,'minimum_interval_seconds' => 30}}, 'App::Koyomi::Config' ) at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Config.pm line 30
2015-07-15T18:11:32 [DEBUG] update schedule at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Schedule.pm line 46
2015-07-15T18:11:33 [DEBUG] (id,user,command) = (3,<NULL>,"echo $(date) alive") at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 49
2015-07-15T18:11:33 [DEBUG] 2015-07-15T18:11:32 Wed at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Schedule.pm line 67
2015-07-15T18:11:33 [INFO] 58165 3 USER=username COMMAND="echo $(date) alive" at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 66
2015-07-15T18:11:33 [INFO] 58165 3 Succeeded. at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 69
2015-07-15T18:11:33 [INFO] 58165 3 ===== STDOUT ===== at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 71
2015-07-15T18:11:33 [INFO] 58165 3 2015年 7月15日 水曜日 18時11分33秒 JST alive\n at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 72
2015-07-15T18:12:03 [DEBUG] no need to update schedule at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Schedule.pm line 42
2015-07-15T18:12:03 [DEBUG] 2015-07-15T18:12:02 Wed at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Schedule.pm line 67
2015-07-15T18:12:03 [INFO] 60859 3 USER=username COMMAND="echo $(date) alive" at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 66
2015-07-15T18:12:03 [INFO] 60859 3 Succeeded. at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 69
2015-07-15T18:12:03 [INFO] 60859 3 ===== STDOUT ===== at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 71
2015-07-15T18:12:03 [INFO] 60859 3 2015年 7月15日 水曜日 18時12分03秒 JST alive\n at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 72
2015-07-15T18:13:01 [DEBUG] no need to update schedule at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Schedule.pm line 42
2015-07-15T18:13:01 [DEBUG] 2015-07-15T18:13:00 Wed at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Schedule.pm line 67
2015-07-15T18:13:01 [INFO] 65829 3 USER=username COMMAND="echo $(date) alive" at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 66
2015-07-15T18:13:01 [INFO] 65829 3 Succeeded. at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 69
2015-07-15T18:13:01 [INFO] 65829 3 ===== STDOUT ===== at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 71
2015-07-15T18:13:01 [INFO] 65829 3 2015年 7月15日 水曜日 18時13分01秒 JST alive\n at /Users/username/.anyenv/envs/plenv/versions/5.20/lib/perl5/site_perl/5.20.2/App/Koyomi/Job.pm line 72
```

```
$ ps alx | ag perl
  502 66539 77162   0  31  0  2423956    212 -      R+   s001    0:00.00 ag perl
  502 58125 99335   0  31  0  2480348  29752 -      S+   s002    0:00.51 perl /Users/username/.anyenv/envs/plenv/versions/5.20/bin/koyomi -c config/default.toml
  502 58165 58125   0   0  0        0      0 -      Z+   s002    0:00.00 (perl5.20.2)
  502 60859 58125   0   0  0        0      0 -      Z+   s002    0:00.00 (perl5.20.2)
  502 65829 58125   0   0  0        0      0 -      Z+   s002    0:00.00 (perl5.20.2)
```

**Comment by papix** [comment]
I tried to resolve this trouble using Parallel::ForkManager: https://github.com/papix/perl5-App-Koyomi/blob/fix/use-parallel-forkmanager/lib/App/Koyomi/Worker.pm
In our environment, This code worked expected.

I hope this helps.

**Comment by key-amb** [comment]
Great! I took a look at your commit.
But I wonder about here: https://github.com/papix/perl5-App-Koyomi/commit/3ddc4aa89b215eb79ee5457303fce8662e8394be#diff-964108aac251e21e7ca74dcbbb50f65eR69

```
+            my $rate = $pm->running_procs / $pm->max_procs;
+            if ($rate > 0.5) {
+                $pm->set_max_procs($pm->max_procs * 2);
+            }
```

Why do you think this block is needed?

And sorry for my late action. I think I can do about koyomi this weekend.

**Comment by key-amb** [comment]
Hi @papix ,

I fixed this by #14 
Please check if it works well.

**Comment by papix** [comment]
I have confirmed that v0.6.0 is running correctly in our environment. 
Thank you very much for fix work!
